### PR TITLE
services: Add Tahoe-LAFS service.

### PR DESCRIPTION
**Pull request by MostAwesomeDude**
Hi! This PR adds tentative (but more or less functional) Tahoe-LAFS service functionality to NixOS.

I have tested this PR by using `nixos-rebuild build-vm` and confirming that the resulting VM has a systemd service which runs Tahoe when started.

I have also tested that a personal machine with NixOS deploys this service as expected using standard `nixos-rebuild switch` incantations.

Not all Tahoe configuration is exposed, but all of the stuff that I cared about is exposed. This should be more or less enough to get a small grid online.
###### Things done:
- [ ] Tested using sandboxing (`nix-build --option build-use-chroot true` or [nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [x] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).
###### More

Fixes issue #<insert id>

cc @<maintainer>

---

_Please note, that points are not mandatory, but rather desired._

### CODE REVIEW

**Comment by jagajaga** [comment]
LGTM

File: nixos/modules/services/network-filesystems/tahoe.nix [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+              enabled = ${if settings.helper.enable then "true" else "false"}
+            '';
+          });
+        # Actually require Tahoe, so that we will have it installed.
+        systemPackages = flip mapAttrsToList cfg.nodes (node: settings:
+          settings.package
+        );
+      };
+      # Open up the firewall.
+      networking.firewall.allowedTCPPorts = flip mapAttrsToList cfg.nodes
```
Comment (copumpkin) [inline nixos/modules/services/network-filesystems/tahoe.nix]: I don't think our general practice is to auto-modify `networking.firewall` when enabling services, instead getting people to manually open up ports separately, as a way to acknowledge that they do want the port externally accessible. I sort of prefer that model, and think it's pretty common in existing NixOS services. Seems like a bigger conversation to have here, but until we've sorted out the bigger question, it's probably best to just stay consistent with the convention and take this out.

Comment (MostAwesomeDude) [inline nixos/modules/services/network-filesystems/tahoe.nix]: I merely copied from the SSH daemon configuration. I'll go with whatever is best practice.

**Comment by fpletz** [comment]
Seems fine to me.

@MostAwesomeDude Should we merge this now into to master to maybe get some feedback/users or do you want add more stuff before doing so?

**Comment by MostAwesomeDude** [comment]
I'm pretty sure that I'm the sole user right now.

I could hack on this more, and get the rest of the configuration sorted out, or we could merge it and then I'll put it to the test in my nixops plan. I'm not really picky; I don't have a preference.

I need sleep, but if this is still open next time I have hacking time, I'll do more work on this.

**Comment by MostAwesomeDude** [comment]
Okay, I'm done for now. This works on a local set of containers (modulo some nixops networking strangeness), so I think that it'll be fine to merge.

**Comment by jagajaga** [comment]
Can you please squash commits?

**Comment by MostAwesomeDude** [comment]
Squashed.

**Pull request merged**